### PR TITLE
Show main install commit in footer

### PR DESCRIPTION
**Pull request by diegopetrucci**
## Summary

Persist the installed checkout commit subject for main-ref installs and show it as a dim suffix in the footer: `TLH main • <commit subject>`. Legacy install state and non-main tracks retain their existing footer behavior.

## Change type

- [x] Installer / wrapper
- [x] Extension / skill / prompt / theme
- [x] Docs
- [ ] CI / release
- [ ] Pin PR (update a `config/default-extensions.json` fork tag)
- [ ] Other

## Validation

**Standard validation:**

```sh
npm run validate
```

Result: passed.

Focused installer, install-state classifier, footer styling, production wiring, legacy fallback, and edge-case tests also pass. Generated runtime outputs are fresh.

## Pre-review checklist

- [x] Change preserves isolation and installer safety invariants:
  - never mutates `~/.pi/agent` (the user's normal Pi configuration)
  - all installer-created Pi commands set `PI_CODING_AGENT_DIR` to the isolated profile directory
  - settings merges are conservative: append missing packages, respect opt-outs, preserve existing isolated-user values, back up before writes
  - does not clobber unmanaged wrapper files without an explicit `--force`
- [x] Relevant tests or smoke checks pass
- [x] Docs and `CHANGELOG.md` updated where a user-visible change warrants it
- [x] Diff contains no secrets, local paths, or unintended generated files

---

## Install this branch

```sh
curl -fsSL https://raw.githubusercontent.com/diegopetrucci/the-last-harness/feature/main-install-commit-footer/install.sh | bash -s -- --ref feature/main-install-commit-footer --track ref
```

## Summary by CodeRabbit

- **New Features**
  - Main-track installations now display the installed commit subject in the footer as `TLH main • <commit subject>`.
  - Commit subjects are sanitized and shown with a dimmed bullet-and-subject suffix.
  - Older installations without commit metadata continue to display `TLH main`.

- **Documentation**
  - Installation documentation and the unreleased changelog now describe the updated main-track footer information.
  - Pinned installation examples now use version `v0.40.0`.

### CODE REVIEW

File: extensions/the-last-harness/footer.ts [hunk 1]
```diff
@@ -262,6 +262,20 @@ export function formatReauthWarningLine(
   return truncateToWidth(countStyled, width, theme.fg("warning", "..."));
 }
 
+function formatTlhInstallNoticeLine(notice: TlhInstallNotice, width: number, theme: Theme): string {
+  const label = formatTlhInstallNoticeTrackLabel(notice);
+  const commitSubject =
+    notice.kind === "ref" && label === "main" && typeof notice.commitSubject === "string"
+      ? sanitizeStatusText(notice.commitSubject)
+      : "";
```
[the remark this answers is not in the document]
Comment (diegopetrucci) [reply extensions/the-last-harness/footer.ts:278]: Fixed in 361c519: the main-subject-only sanitizer removes all Unicode Cc controls, including ESC and C1, before footer rendering. The generated JavaScript mirror and focused ESC/C1 tests are included.

File: extensions/the-last-harness/footer.ts [hunk 2]
```diff
@@ -262,6 +262,20 @@ export function formatReauthWarningLine(
   return truncateToWidth(countStyled, width, theme.fg("warning", "..."));
 }
 
+function formatTlhInstallNoticeLine(notice: TlhInstallNotice, width: number, theme: Theme): string {
+  const label = formatTlhInstallNoticeTrackLabel(notice);
+  const commitSubject =
+    notice.kind === "ref" && label === "main" && typeof notice.commitSubject === "string"
+      ? sanitizeStatusText(notice.commitSubject)
```
[the remark this answers is not in the document]
Comment (diegopetrucci) [reply extensions/the-last-harness/footer.ts]: Fixed in 361c519: persisted main commit subjects now strip every Unicode Cc control before rendering, while preserving CR/LF/tab whitespace normalization. ESC and C1 regression tests were added.

File: tests/tlh-install-state.test.mjs [hunk 3]
```diff
@@ -71,6 +71,14 @@ test("tlh-install-state dry-run reports the write without creating install-state
   assert.equal(existsSync(fixture.statePath), false);
 });
 
+test("tlh-install-state accepts a hyphen-leading installed commit subject", () => {
+  const fixture = tempFixture();
```
[the remark this answers is not in the document]
Comment (diegopetrucci) [reply tests/tlh-install-state.test.mjs]: Fixed in 361c519: tempFixture now accepts the test context and registers recursive forced cleanup once at the helper level; all four call sites use it.
